### DISCUSSION TITLE
Design discussion: Allow using the same reference block when consuming notes.

### DISCUSSION THREAD

**Discussion by hackaugusto** [0xMiden/node#472] [Ideas]
ref: https://discord.com/channels/893151281848406016/1148602549058949120/1209932644423303289

tl;dr: the user can use the same block header as a reference when creating output notes. this is used by the faucet to avoid doing `sync` calls. this issue is to discuss the possibility of doing the same for input notes.

**Comment by bobbinth** [comment]
The main challenge here is that for input notes, the transaction kernel needs to prove that the note is included in the chain MMR. Two potential way to get around this:

1. The inclusion proof for notes is delegated to the network. This would imply that we give up privacy as to which note is consumed by the transaction.
2. The inclusion path is included with the note. This is kind of like doing a sync but without using the sync endpoint.

**Comment by hackaugusto** [comment]
> The inclusion path is included with the note. This is kind of like doing a sync but without using the sync endpoint.

If we have Alice sending a note to Bob, and this method is used. I think if Bob trusts the sequencer to validate the inclusion proof in the send proven transaction. That should be enough to check the inclusion proof is valid without a sync call.

One cool thing that could be done too, is that Alice sends a ZK proof to Bob, that proves the authentication path against a block from L1, and then Bob has only to fetch the block header.
